### PR TITLE
DAOS-5567 vos: Fix issue with negative entry initialization

### PR DESCRIPTION
**Pull request by jolivier23**
When initializing a new negative entry timestamp for a key,
only the low timestamp of the parent should be used.  This
timestamp logically represents a read to the non-existent key.
The high timestamp represents the high timestamp of reads to
any entity under the key but this would not include the new
negative entry because it doesn't exist.

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: src/common/tests_lib.c [hunk 1]
```diff
@@ -51,7 +51,8 @@ dts_oid_gen(uint16_t oclass, uint8_t ofeats, unsigned seed)
 	/* generate a unique and not scary long object ID */
 	oid.lo	= obj_id_gen++;
 	oid.lo	|= hdr;
-	oid.hi	= rand() % 100;
+	/** Mod by large 32-bit prime */
+	oid.hi	= rand() % 2147483647;
```
Comment (jolivier23) [inline src/common/tests_lib.c:55]: Oops.  I could revert this change.  On the other hand, it will make the value more unique.

File: src/common/tests_lib.c [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /**
- * (C) Copyright 2015-2019 Intel Corporation.
+ * (C) Copyright 2015-2020 Intel Corporation.
```
Comment (jolivier23) [inline src/common/tests_lib.c:2]: I reverted that change as it caused issues but this file has changed this year

**Review by Nasf-Fan — APPROVED** [review]
The patch looks reasonable for me.

**Review by gnailzenh — APPROVED** [review]
